### PR TITLE
Friend icons

### PR DESCRIPTION
**Pull request by mlytvynchuk**

### CODE REVIEW

File: app/src/main/res/drawable/ic_not_yet_friends.xml [hunk 1]
```diff
@@ -0,0 +1,106 @@
+<!--
```
Comment (saimonsaret) [inline app/src/main/res/drawable/ic_not_yet_friends.xml:1]: Where this is coming from?

File: app/src/main/java/com/google/droidjump/MainActivity.java [hunk 2]
```diff
@@ -150,6 +175,29 @@ protected void onResume() {
         }
     }
 
+    public Task<AnnotatedData<PlayerBuffer>> loadFriendNames() {
+        return getPlayersClient().loadFriends(GameConstants.FRIENDS_PER_PAGE, /* forceReload = */ false)
+                .addOnSuccessListener(buffer -> {
+                    playerBuffer = buffer.get();
+                    try {
```
Comment (saimonsaret) [inline app/src/main/java/com/google/droidjump/MainActivity.java:182]: What can fail there and why not catching an exception that is thrown?

File: app/src/main/java/com/google/droidjump/LeaderboardScoresFragment.java [hunk 3]
```diff
@@ -85,7 +88,7 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
         View rootView = inflater.inflate(R.layout.leaderboard_scores_screen, container, /* attachToRoot= */ false);
         ImageManager.create(activity).loadImage((ImageView) rootView.findViewById(R.id.leaderboard_avatar), leaderboard.getIconImageUri());
         ((TextView) rootView.findViewById(R.id.leaderboards_title)).setText(leaderboard.getDisplayName());
-        adapter = new LeaderboardsScoresAdapter(scores, activity);
+        adapter = new LeaderboardsScoresAdapter(scores, activity, activity.getFriendNames(), activity.isFriendListAccess());
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/LeaderboardScoresFragment.java:91]: isFriendListAccess -- maybe hasAccessToFriendsList?

File: app/src/main/java/com/google/droidjump/MainActivity.java [hunk 4]
```diff
@@ -150,6 +175,29 @@ protected void onResume() {
...
+                        while (DataBufferUtils.hasNextPage(playerBuffer)) {
+                            getPlayersClient().loadMoreFriends(GameConstants.FRIENDS_PER_PAGE).addOnSuccessListener(data -> {
+                                playerBuffer = data.get();
+                            });
+                        }
+                    } finally {
+                        friendNames.clear();
+                        for (Player player : playerBuffer) {
+                            String friendName = player.freeze().getDisplayName();
+                            friendNames.add(friendName);
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/MainActivity.java:192]: how is the order of the friends name preserved?

File: app/src/main/res/drawable/ic_not_yet_friends.xml (hunk 1 above)
Comment (mlytvynchuk) [reply app/src/main/res/drawable/ic_not_yet_friends.xml:1]: https://developers.google.com/games/services/checklist

File: app/src/main/java/com/google/droidjump/MainActivity.java (hunk 2 above)
Comment (mlytvynchuk) [reply app/src/main/java/com/google/droidjump/MainActivity.java:182]: Fixed. That how it was coded on the project from play game services android samples on Github.

File: app/src/main/java/com/google/droidjump/MainActivity.java (hunk 4 above)
Comment (mlytvynchuk) [reply app/src/main/java/com/google/droidjump/MainActivity.java:192]: Alphabetically.

**Review by redls — APPROVED** [review]

**Pull request merged**